R/package.R

```diff
@@ -50,6 +50,10 @@
 argufy <- function(fun, ...) {
   if (!is.function(fun)) stop("'fun' must be a function")
 
+  # these statements are needed to get S4 functions to work properly
+  was_s4 <- isS4(fun)
+  old_attributes <- attributes(fun)
+
```

Comment:
Also, this is not quite good, because `attr(, "srcref")` is copied as well, and it is used for printing the (new) function. I am not entirely sure what to do with this, though. If we don't copy, then the new function will have no `srcref`, which might be a problem sometimes, no? But we also do not want to reparse the function....